tests/common/include/tests/common/tools_stepper.inl

```diff
@@ -217,3 +219,97 @@ TEST(ALGEBRA_PLUGIN, rk_stepper) {
         }
     }
 }
+
+// This tests the covariance transport in rk stepper
+TEST(ALGEBRA_PLUGIN, covariance_transport) {
+
+    using vector3 = vector3<scalar>;
+    using point3 = point3<scalar>;
+    using matrix_operator = standard_matrix_operator<scalar>;
+    using mag_field_t = constant_magnetic_field<>;
+    using crk_stepper_t =
+        rk_stepper<mag_field_t, free_track_parameters, constrained_step<>>;
+
+    // Generate track starting point
+    point3 pos{0., 0., 0.};
+    scalar time = 0.;
+    vector3 mom{0.01, 0.01, 0.01};
+    scalar q = -1.;
+
+    free_track_parameters vertex(pos, time, mom, q);
+
+    // Fill covariance matrix
+    auto covariance0 =
+        matrix_operator().template zero<e_free_size, e_free_size>();
+
+    matrix_operator().element(covariance0, e_free_pos0, e_free_pos0) = 0.1;
+    matrix_operator().element(covariance0, e_free_pos1, e_free_pos1) = 0.1;
+    matrix_operator().element(covariance0, e_free_pos2, e_free_pos2) = 0.1;
+    matrix_operator().element(covariance0, e_free_dir0, e_free_dir0) = 0.1;
+    matrix_operator().element(covariance0, e_free_dir1, e_free_dir1) = 0.1;
+    matrix_operator().element(covariance0, e_free_dir2, e_free_dir2) = 0.;
+    matrix_operator().element(covariance0, e_free_qoverp, e_free_qoverp) = 0.;
+
+    vertex.set_covariance(covariance0);
+
+    // magnetic field
+    vector3 B{0, 0, 10. * unit_constants::T};
```

Comment:
That's quite a strong b-field :)

Reply:
Agree. I just needed to reduce the path length to reduce the error. As I said above:
> I could not figure out a good error tolerance expression between two jacobians, therefore, I just set it as (path_length * epsilon) but we will need to find out better expression...

Once we find a good expression I might be able to try smaller B field without tuning tolerance value...